app/javascript/app/components/emission-pathways-scenario-table/emission-pathways-scenario-table-selectors.js

```diff
@@ -17,6 +18,28 @@ const getIndicatorsData = createSelector(getIdData, data => {
   return data.indicators || null;
 });
 
+export const getCategories = createSelector(getIndicatorsData, data => {
+  if (!data) return null;
+  return ['All']
+    .concat(uniq(data.map(i => i.category)))
+    .map(c => ({ value: c, label: c }));
+});
+
+const getSelectedCategory = createSelector(
+  [state => state.categorySelected, getCategories],
+  (selected, categories = []) => {
+    if (categories.length > 0) {
+      return selected || 'All';
+    }
+    return null;
+  }
+);
+
+export const getSelectedCategoryObject = createSelector(
+  getSelectedCategory,
+  selected => ({ value: selected, label: selected } || null)
```

Comment:
You already have the structure from the map. You shouldn't need to remap this. It should be a simple filter.